### ISSUE TITLE
`rustup update` results in "rustfmt is unavailable for download"

### ISSUE THREAD

**Issue by Hi-Angel** [rust-lang/rustup#1958]
**Problem**

`rustup update` results in an error:

```
$ rustup update
info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu'
info: latest update on 2019-07-29, rust version 1.38.0-nightly (4560cb830 2019-07-28)
error: component 'rustfmt' for target 'x86_64-unknown-linux-gnu' is unavailable for download for channel 'nightly'
If you require these components, please install and use the latest successful build version,
which you can find at https://rust-lang-nursery.github.io/rust-toolstate.

After determining the correct date, install it with a command such as:

    rustup toolchain install nightly-2018-12-27

Then you can use the toolchain with commands such as:

    cargo +nightly-2018-12-27 build

  nightly-x86_64-unknown-linux-gnu update failed - rustc 1.32.0-nightly (d31157190 2018-12-01)
```

I can't seem to remove `rustfmt` either:

```
$ rustup component remove rustfmt
error: toolchain 'nightly-x86_64-unknown-linux-gnu' does not contain component 'rustfmt' for target 'x86_64-unknown-linux-gnu'; did you mean 'rust-std'?
```

I was advised on IRC to report a bug.

**Steps**

For me it's just running `rustup update`, unfortunately I don't know how to reproduce it in general.

**Possible Solution(s)**

Supposedly, running `rustup toolchain remove nightly && rustup toolchain install nightly` should fix it, but I didn't try that yet in case the current rustup state may be useful for the report.

**Notes**

* Output of `rustup --version`:

      rustup 1.18.3 (2019-05-23)
* Output of `rustup show`:

    ```
    Default host: x86_64-unknown-linux-gnu

    installed toolchains
    --------------------

    nightly-2018-01-02-x86_64-unknown-linux-gnu
    nightly-2018-02-08-x86_64-unknown-linux-gnu
    nightly-2018-03-01-x86_64-unknown-linux-gnu
    nightly-x86_64-unknown-linux-gnu (default)

    installed targets for active toolchain
    --------------------------------------

    i686-unknown-linux-gnu
    x86_64-unknown-linux-gnu

    active toolchain
    ----------------

    nightly-x86_64-unknown-linux-gnu (default)
    rustc 1.32.0-nightly (d31157190 2018-12-01)
    ```

**Comment by pBeta00n** [comment]
1551950392TCL Communication Press Release_FINAL-.pdf_CES 2019_

**Comment by aleksmelnikov** [comment]
https://github.com/rust-lang/rustup.rs/issues/1957

**Comment by kinnison** [comment]
@aleksmelnikov That's not *quite* what I take from this issue.

To me, it reads as:

* **GIVEN** a working nightly which includes `rustfmt`
* **WHEN** I run `rustup update` to a nightly which does not include `rustfmt`
* **THEN** the update fails
* **AND** I can no longer remove `rustfmt`

That implies that the update has replaced the channel manifest file, but not completed the update.  This would be a bad thing ™ and needs investigating.

**Comment by kinnison** [comment]
Can you confirm if this still happens with Rustup 1.20.2 ?

**Comment by Hi-Angel** [comment]
Sorry, 2 weeks ago I needed to contribute [to a Rust project](https://github.com/mookid/diffr/pull/32), so I had to do `rustup toolchain remove nightly && rustup toolchain install nightly` to make it work. So I can't test the reported problem anymore. Feel free to close it if you don't think it's useful to keep opened.

**Comment by kinnison** [comment]
Okay thanks, I think it's fixed in 1.20 so I'm going to close this.  Please open a fresh issue if you have a new example of this occurring.
